doc/release/1.17.0-notes.rst

```diff
@@ -42,6 +42,18 @@ In the future it will not be possible to switch the writeable flag to ``True``
 from python.
 This deprecation should not affect many users since arrays created in such
 a manner are very rare in practice and only available through the NumPy C-API.
+Writing to the result of `numpy.broadcast_array` will warn
+----------------------------------------------------------
+
+Commonly `numpy.broadcast_array` returns a writeable array with internal
+overlap, making it unsafe to write to. A future version will set the
+``writeable`` flag ``False``, and require users to manually set it ``True`` if
+they are sure that is what they want to do. Writing to it will now emit a
+deprecation warning with instructions to set the ``wrietable`` flag ``True``.
+This creates a confusing situation where even though the ``writeable`` flag is
```

Comment:
I think this sentence only confuses the reader more! Maybe instead  "Note that if one were to inspect the flag before setting it, one would, perhaps confusingly, find it would already be `True`. Explicitly setting it, though, as one will need to do in future versions, clears an internal flag that is used to produce the deprecation warning."

Reply:
adopted